spec/features/users/user_create_account_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+feature 'user can create an account' do
+  scenario 'user submits information and this user is added to database' do
+
+    visit new_user_path
+
+    fill_in 'user[name]', with: 'test'
+    fill_in 'user[email]', with: 'test@test.com'
+    fill_in 'user[password]', with: 'password'
+    fill_in 'user[password_confirmation]', with: 'password'
+    expect { click_button 'Create Account'}.to change(User, :count).by(1)
+
+    expect(current_path).to eq user_path(User.last)
+    expect(page).to have_content("Welcome..#{User.last.name}..")
+  end
+
+  scenario 'user submits incomplete information and is redirected to the new user page' do
+
+    visit new_user_path
+
+    fill_in 'user[name]', with: ''
+    fill_in 'user[email]', with: 'test@test.com'
+    fill_in 'user[password]', with: ''
+    fill_in 'user[password_confirmation]', with: ''
+
+    expect { click_button 'Create Account'}.to change(User, :count).by(0)
+
+    expect(current_path).to eq new_user_path
+  end
+
+  scenario 'user already exists and current user is redirected to the new user page' do
+    User.create(name: 'test', email: 'test@test.com', password_digest: 'password', password_confirmation: 'password')
+
+      visit new_user_path
+
+        fill_in 'user[name]', with: ''
+        fill_in 'user[email]', with: 'test@test.com'
+        fill_in 'user[password]', with: ''
+        fill_in 'user[password_confirmation]', with: ''
+
+        expect { click_button 'Create Account'}.to change(User, :count).by(0)
+
+        expect(current_path).to eq new_user_path
```

Comment:
This is super interesting. I'm wondering, does it actually send you back to `/users` or does it just display the new users template, but show `/users` in the nav bar/as the `current_path`? What I believe is happening with render here is that it is displaying the path of the request that was used to generate it. So, in your example, a `POST` request sent to `/users` that uses a `render` in the controller action will show up in the browser's nav bar as `/users` (even if it's showing the new user form).

I wonder if you take out this expectation and instead put something about what you expect to see on the page ("Sorry, user not successfully created", or some such thing) if it would pass. Let me know!